config/config.go

```diff
@@ -24,6 +25,13 @@ type HAproxyConfig struct {
 	UseHostnames bool   `envconfig:"USE_HOSTNAMES"`
 }
 
+type EnvoyConfig struct {
+	UseGRPCAPI   bool   `envconfig:"USE_GRPC_API" default:"false"`
+	BindIP       string `envconfig:"BIND_IP" default:"192.168.168.168"`
+	UseHostnames bool   `envconfig:"USE_HOSTNAMES"`
+	GRPCPort     string `envconfig:"GRPC_PORT" default:"8502"` // The default gRPC port used by Consul
```

Comment:
This Envoy gRPC API needs to spin up a gRPC server in the Sidecar process, so Sidecar will need to be able to bind to this port. Please let me know if another default value is more appropriate. I saw Consul used `8502` for this and Contour uses [`8001`](https://github.com/projectcontour/contour/blob/cada619f7dd7f0cb1e82bb3f0e8c7f90d8c43cac/cmd/contour/servecontext.go#L119).

Reply:
I'd probably pick 7776 so that it's grouped right next to our other ports. The other Sidecar ecosystem services are like 7778 and 7779.